core/src/Kokkos_CopyViews.hpp

```diff
@@ -871,381 +706,55 @@ struct ViewRemap<DstType, SrcType, ExecSpace, 1> {
-        p_type ext0(0, std::min(dst.extent(0), src.extent(0)));
-        p_type ext1(0, std::min(dst.extent(1), src.extent(1)));
-        using sv_adapter_type =
-            CommonSubview<DstType, SrcType, 3, p_type, p_type, Kokkos::ALL_t>;
-        sv_adapter_type common_subview(dst, src, ext0, ext1, Kokkos::ALL);
-        view_copy(exec_space..., common_subview.dst_sub,
-                  common_subview.src_sub);
-      } else {
-        p_type ext0(0, std::min(dst.extent(0), src.extent(0)));
-        p_type ext1(0, std::min(dst.extent(1), src.extent(1)));
-        p_type ext2(0, std::min(dst.extent(2), src.extent(2)));
-        using sv_adapter_type =
-            CommonSubview<DstType, SrcType, 3, p_type, p_type, p_type>;
-        sv_adapter_type common_subview(dst, src, ext0, ext1, ext2);
-        view_copy(exec_space..., common_subview.dst_sub,
-                  common_subview.src_sub);
-      }
-    }
-  }
-};
-
-template <class DstType, class SrcType, class ExecSpace>
-struct ViewRemap<DstType, SrcType, ExecSpace, 4> {
+template <class DstType, class SrcType, std::size_t... I>
+auto create_common_subview_first_match(const DstType& dst, const SrcType& src,
+                                       std::index_sequence<I...>) {
   using p_type = Kokkos::pair<int64_t, int64_t>;
+  CommonSubview common_subview(
+      dst, src, Kokkos::ALL,
+      (p_type(0, std::min(dst.extent(I + 1), src.extent(I + 1))))...);
+  return common_subview;
+}
 
-  template <typename... OptExecSpace>
-  ViewRemap(const DstType& dst, const SrcType& src,
-            const OptExecSpace&... exec_space) {
-    static_assert(
-        sizeof...(OptExecSpace) <= 1,
-        "OptExecSpace must be either empty or be an execution space!");
-
-    if (dst.extent(0) == src.extent(0)) {
-      if (dst.extent(3) == src.extent(3)) {
-        p_type ext1(0, std::min(dst.extent(1), src.extent(1)));
-        p_type ext2(0, std::min(dst.extent(2), src.extent(2)));
-        using sv_adapter_type =
-            CommonSubview<DstType, SrcType, 4, Kokkos::ALL_t, p_type, p_type,
-                          Kokkos::ALL_t>;
-        sv_adapter_type common_subview(dst, src, Kokkos::ALL, ext1, ext2,
-                                       Kokkos::ALL);
-        view_copy(exec_space..., common_subview.dst_sub,
-                  common_subview.src_sub);
-      } else {
-        p_type ext1(0, std::min(dst.extent(1), src.extent(1)));
-        p_type ext2(0, std::min(dst.extent(2), src.extent(2)));
-        p_type ext3(0, std::min(dst.extent(3), src.extent(3)));
-        using sv_adapter_type =
-            CommonSubview<DstType, SrcType, 4, Kokkos::ALL_t, p_type, p_type,
-                          p_type>;
-        sv_adapter_type common_subview(dst, src, Kokkos::ALL, ext1, ext2, ext3);
-        view_copy(exec_space..., common_subview.dst_sub,
-                  common_subview.src_sub);
-      }
-    } else {
-      if (dst.extent(7) == src.extent(7)) {
```

Comment:
This was a bug. It doesn't make sense to compare extent(7) for a rank 4 View. It should have been extent(3).